components/expenses/graphql/fragments.js

```diff
@@ -174,6 +174,7 @@ export const expensePageExpenseFieldsFragment = gqlV2/* GraphQL */ `
       type
       isAdmin
       location {
+        id
```

Comment:
This line alone fixes https://github.com/opencollective/opencollective/issues/5627. Without it, Apollo was triggering a full refetch of the expense page query.